test/fixtures/transpiler/amd/Factory_BasicModule.js

```diff
@@ -2,6 +2,9 @@ sap.ui.define(["sap/ui/core/mvc/Controller", "sap/ui/core/UIComponent", "sap/ui/
 	function (Controller, UIComponent, History, Button) {
 	"use strict";
 
+	/**
+	 * @class JSDoc comments should be removed, as it might have negative impact on the type detection
+	 */
```

Comment:
If TS parses and extracts comments the way JSDco does, then this comment would be totally ignored. JSDoc ignores comments above return statements

Reply:
That's true. Right now we ignore JSDoc comments everywhere, but in this case it wouldn't matter.